rust_src/src/eval.rs

```diff
@@ -1334,4 +1330,158 @@ pub fn condition_case(args: LispCons) -> LispObject {
     unsafe { internal_lisp_condition_case(var, bodyform, handlers) }
 }
 
+// Eval a sub-expression of the current expression (i.e. in the same lexical scope).
+#[no_mangle]
+pub extern "C" fn eval_sub(form: LispObject) -> LispObject {
+    if let Some(symbol) = form.as_symbol() {
+        let env = unsafe { globals.Vinternal_interpreter_environment };
+
+        // Look up its binding in the lexical environment.
+        // We do not pay attention to the declared_special flag here, since we
+        // already did that when let-binding the variable.
+        let lex_binding = env.map_or(Qnil, |e| assq(form, e));
+
+        if let Some((_, cdr)) = lex_binding.into() {
```

Comment:
In general I'd like to start using more descriptive names for cons elements.

Reply:
Agreed. It is sometimes hard to name them.